test/test_web.py

```diff
@@ -0,0 +1,143 @@
+
+
+def test_search_urls_multiple_urls():
+    urls = list(search_urls('http://a.com/ http://b.com/'))
+    assert len(urls) == 2, 'Must find 2 URLs, found %d' % len(urls)
+    assert 'http://a.com/' in urls
+    assert 'http://b.com/' in urls
+
+
+def test_search_urls_multiple_urls_with_text():
+    urls = list(
+        search_urls('before http://a.com/ between http://b.com/ after'))
+    assert len(urls) == 2, 'Must find 2 URLs, found %d' % len(urls)
+    assert 'http://a.com/' in urls
+    assert 'http://b.com/' in urls
+
+
+def test_search_urls_multiple_urls_unique():
+    urls = list(search_urls('http://a.com/ http://b.com/ http://a.com/'))
+    assert len(urls) == 2, 'Must find 2 URLs, found %d' % len(urls)
+    assert 'http://a.com/' in urls
+    assert 'http://b.com/' in urls
+
+
+def test_search_urls_multiple_urls_unique_keep_ordering():
+    urls = list(
+        search_urls('http://a.com/ http://c.com/ http://b.com/ http://a.com/'))
+    assert len(urls) == 3, 'Must find 3 URLs, found %d' % len(urls)
+    assert 'http://a.com/' in urls
+    assert 'http://b.com/' in urls
+    assert 'http://c.com/' in urls
+    assert urls == [
+        'http://a.com/',
+        'http://c.com/',
+        'http://b.com/',
+    ]
+
+
+def test_search_urls_exclusion_char():
+    # assert url is excluded
+    urls = list(search_urls('!http://example.com', exclusion_char='!'))
+    assert not urls, 'Must not find URL, found %d' % len(urls)
+
+    # assert the other url is not excluded
+    urls = list(
+        search_urls('http://b.com !http://a.com', exclusion_char='!'))
+    assert len(urls) == 1, 'Must find 1 URL, found %d' % len(urls)
+    assert 'http://b.com' in urls
+
+    # assert the order of appearance does not matter
+    urls = list(
+        search_urls('!http://a.com http://b.com', exclusion_char='!'))
+    assert len(urls) == 1, 'Must find 1 URL, found %d' % len(urls)
+    assert 'http://b.com' in urls
+
+
+def test_search_urls_exclusion_char_with_text():
+    urls = list(
+        search_urls(
+            'before !http://a.com between http://b.com after',
+            exclusion_char='!')
+    )
+    assert len(urls) == 1, 'Must find 1 URL, found %d' % len(urls)
+    assert 'http://b.com' in urls
+
+
+def test_search_urls_exclusion_char_only_once():
+    # assert only the instance excluded is excluded
+    # ie. that it is not a global exclude, otherwise that would return 1 url
+    urls = list(
+        search_urls(
+            '!http://a.com http://a.com http://b.com',
+            exclusion_char='!')
+    )
+    assert len(urls) == 2, 'Must find 1 URL, found %d' % len(urls)
+    assert 'http://a.com' in urls
+    assert 'http://b.com' in urls
+
+
+TRAILING_CHARS = list('.,?!\'":;')
+ENCLOSING_PAIRS = [('(', ')'), ('[', ']'), ('{', '}'), ('<', '>')]
+
+
+@pytest.mark.parametrize('trailing_char', TRAILING_CHARS)
+def test_trim_url_remove_trailing_char(trailing_char):
+    test_url = 'http://exemple.com/'
```

Comment:
It doesn't actually _matter_ in the slightest, but above here we have `example.com` and from here to the end of the file it's `exemple.com` instead. 😕

Reply:
lol, French being bad at English :D